sys/net/gnrc/network_layer/ipv6/nib/nib.c

```diff
@@ -796,20 +811,23 @@ static void _send_delayed_nbr_adv(const gnrc_netif_t *netif,
     reply_flags |= NDP_NBR_ADV_FLAGS_O;
 #endif  /* GNRC_NETIF_L2ADDR_MAXLEN > 0 */
     /* discard const qualifier */
-    nbr_adv = gnrc_ndp_nbr_adv_build(tgt, reply_flags, extra_opts);
-    if (nbr_adv == NULL) {
-        DEBUG("nib: No space left in packet buffer. Not replying NS");
-        gnrc_pktbuf_release(extra_opts);
+    pkt = gnrc_ndp_nbr_adv_build(tgt, reply_flags, payload);
+    if ((payload = _check_release_pkt(pkt, payload)) == NULL) {
         return;
     }
-    nce = _nib_onl_get(tgt, netif->pid);
-    if ((nce != NULL) && (nce->mode & _NC)) {
-        /* usually this should be the case, but when NCE is full, just
-         * ignore the sending. Other nodes in this anycast group are
-         * then preferred */
-        _evtimer_add(nce, GNRC_IPV6_NIB_SND_NA, &nce->snd_na,
-                     random_uint32_range(0, NDP_MAX_ANYCAST_MS_DELAY));
+    pkt = gnrc_ipv6_hdr_build(payload, NULL, &ipv6_hdr->src);
+    if ((payload = _check_release_pkt(pkt, payload)) == NULL) {
+        return;
+    }
+    ((ipv6_hdr_t *)payload->data)->hl = 255;
```

Comment:
why `255` and not the configured default for the interface?

Reply:
Because NAs [must have their hop-limit set to 255](https://tools.ietf.org/html/rfc4861#section-4.4) to ensure they are link-local. NAs with any other hop-limit value [are to be ignored silently](https://tools.ietf.org/html/rfc4861#section-7.12).

Reply:
Maybe this value could be provided as a define with some documentation ?

Reply:
Sure, but since I use this magic number also in `gnrc_ndp_*_send()` I'd rather do this in a follow-up, so I don't need to touch these functions in a PR that needs to be backported.

Reply:
>Sure, but since I use this magic number also in `gnrc_ndp_*_send()` I'd rather do this in a follow-up, so I don't need to touch these functions in a PR that needs to be backported.

:+1: